src/client/components/hud/Initiative.tsx

```diff
@@ -252,17 +259,88 @@ export function InitiativeHUD() {
 
   const focusTokenOnTurnStart = useRRSettings()[0].focusTokenOnTurnStart;
 
+  const updateConcentration = useCallback(
+    (
+      character: RRCharacter,
+      updater: React.SetStateAction<RRCharacter["currentlyConcentratingOn"]>
+    ) => {
+      dispatch((state) => {
+        const old =
+          state.characters.entities[character.id]?.currentlyConcentratingOn;
+
+        if (old === undefined) return [];
+        const newConcentration =
+          typeof updater === "function" ? updater(old) : updater;
+
+        return {
+          actions: [
+            characterUpdate({
+              id: character.id,
+              changes: { currentlyConcentratingOn: newConcentration },
+            }),
+          ],
+          optimisticKey: "concentration",
+          syncToServerThrottle: 0,
+        };
+      });
+    },
+    [dispatch]
+  );
+
+  const alertPlayer = useCallback(
+    async (characterName: string, concentrationObject: string) => {
+      await alert(
+        `${characterName}'s concentration on ${concentrationObject} has expired.`
+      );
+    },
+    [alert]
+  );
+
   useEffect(() => {
     if (
-      focusTokenOnTurnStart &&
       currentRow &&
       currentRow.id !== lastRowId.current &&
       currentRow.type === "character"
     ) {
-      setSelection(currentRow.characterIds);
+      if (focusTokenOnTurnStart) setSelection(currentRow.characterIds);
+      currentRow.characterIds.forEach((characterId) => {
+        const character = charactersRef.current.entities[characterId];
+        if (character?.currentlyConcentratingOn) {
+          updateConcentration(character, {
+            ...character.currentlyConcentratingOn,
+            roundsLeft: character.currentlyConcentratingOn.roundsLeft - 1,
+          });
```

Comment:
I think this code should be moved into an `else` branch of the `if` statement below. That would make it clearer why the condition is actually `<= 1`, and not `< 1`, as you might expect with the current code.